### ISSUE TITLE
[FE] My VA: 508-defect-4 [COGNITION]: "Use my VA health" opens in a new tab without warning the user

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#29043]
# [508-defect-4](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-4)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## Details

Currently, the "Use my VA health" link for Cerner users opens a new tab without warning the user.

WCAG G200: Opening new windows and tabs from a link only when necessary

> The objective of this technique is to limit the use of links or buttons that open new windows or tabs within Web content. In general, it is better not to open new windows and tabs since they can be disorienting for people, especially people who have difficulty perceiving visual content. However there are some situations where it is preferable from an accessibility perspective to open a new window or tab. Here are two such situations:

Opening a page containing context-sensitive information, such as help instructions, or an alternate means of completing a form, such as a calendar-based date picker, will significantly disrupt a multi-step workflow, such as filling in and submitting a form, if the page is opened in the same window or tab.

The user is logged into a secured area of a site, and following a link to a page outside of the secured area would terminate the user's logon. In this case opening external links in an external window allows the user to access such references while keeping their login active in the original window.

It is recommended that when links are opened to a new window, there is advance warning.

## Acceptance Criteria

- [ ] **Preferred**: "Use my VA health" link does not open in a new tab (remove _blank) OR
- [ ] Users are given an advance warning that "Use my VA health" will open in a new tab by means of:
  - [ ] **Preferred**: including "opens in a new tab" in the link text e.g. "Use my VA health (opens in new tab)" OR
  - [ ] Using an `aria-label` to append opens in new tab for screen reader users

## Steps to Recreate

1. Log in as Cerner user and go to My VA
2. Click the "Use my VA health" link
3. Verify it opens in a new tab

## Proposed Solution (if known)

See AC

## WCAG or Vendor Guidance (optional)

* [G200: Opening new windows and tabs from a link only when necessary](https://www.w3.org/TR/WCAG20-TECHS/G200.html)
* [Smashing Magazine: Should Links Open In New Windows?](https://www.smashingmagazine.com/2008/07/should-links-open-in-new-windows/)

**Comment by ajakabcin** [comment]
Hey @joshkimux we had a question about this ticket - isn't this standard behavior? Just curious if this is a new a11y requirement or if we need to look into this issue elsewhere. Thanks in advance for the assistance!

cc @erikphansen and @tmitchellgcio

**Comment by joshkimux** [comment]
@ajakabcin Yes and no-- I can explain 😄 

This is a new _recommendation_ we're making to stay in accordance with WCAG best practices, specifically:
- [G200: Opening new windows and tabs from a link only when necessary](https://www.w3.org/TR/WCAG20-TECHS/G200.html)
- [G201: Giving users advanced warning when opening a new window](https://www.w3.org/TR/WCAG20-TECHS/G201.html)

For more info on this, I'd recommend [checking out our presentation on links](https://docs.google.com/presentation/d/14AwUoFOg_Ix1Txl9oePtPD1ntOpylBWwBJMYq9f5vmo/edit?usp=sharing) which goes into more detail on why opening in a new tab can be disorientating for folks who use magnification tools and screen readers.

**Comment by ajakabcin** [comment]
Thanks @joshkimux! Ok so if we make this change here, we probably would then need to do an audit of sorts elsewhere to see where windows/tabs open without warning, for consistency?

**Comment by joshkimux** [comment]
@ajakabcin Agreed-- I'll look into creating an online document to track issues we find elsewhere like we did with buttons vs. links! 👍

**Comment by erikphansen** [comment]
@joshkimux I feel like someone else (not sure who) would have to weigh in on no longer opening external links in new tabs/windows since that has been such an established "best" practice that is clearly not actually the best practice. I'm happy to kill the `"_blank"` behavior, but want to make sure this change doesn't upset other people in the org. Or do you think I should just do it?

**Comment by joshkimux** [comment]
@erikphansen I can clarify a bit in terms of my intention; the goal isn't to end `_blank` behavior, because there are use cases where it is viable. I'm hoping to propose more specific guidelines around when it should be used, and in those cases, how to provide a heads up to folks before they click on it. 

In this case-- depending on the context-- there's two proposed recommendations in the ticket. One simply involves removing the `_blank` behavior, but the other alternatives suggest either adding "opens in new tab" to the text link while keeping the `_blank` behavior.

That being said, is there anyone you'd recommend me to reach out to ahead of time to get thoughts on this? More than happy to have that conversation before pushing this forward 😄

**Comment by erikphansen** [comment]
@joshkimux sure I follow, I was just leaning towards doing your **preferred** fix because it does make sense :) Leaving VA.gov isn't going to have the side effect of signing you out or anything like that.

I'd ask Mikki about this because she used to do a lot of the IA stuff and this sort of falls under that. She might have a lead on the correct person or team to raise this to.

**Comment by joshkimux** [comment]
@erikphansen Awesome! If there isn't going to be any undesired side effects, taking out `_blank` and making it a regular old link would be preferred 🙏 

Also, thanks for recommending Mikki-- I should have thought about that sooner! Appreciate the tip.

**Comment by ajakabcin** [comment]
ok @erikphansen so for this ticket we'll go with the first recommendation in the AC:

> Preferred: "Use my VA health" link does not open in a new tab (remove _blank)

and then @joshkimux is going to follow up with @mnorthuis on the underlying concern and potential broader solution(s)

**Comment by mnorthuis** [comment]
Hi all, thanks for the tag!  Catching up on the convo, here's my thoughts...

We have always had some inconsistencies across the site on this, and probably need to revisit it.  Typically we only opened a new tab if 1) clicking the link will take a user out of a process and risk losing data (i.e. filling out a form), 2) opening the link in the same window would be confusing, or 3) it is an external link.  

For the record, we generally treat the subdomains of va.gov as staying within the realm of va.gov and not external. However, the Cerner site does not provide easy access back to VA.gov via a header/footer link or nav option like most subdomains do, and users have to log in (again).  Those aspects do make it feel like an external site, and the additional login could be confusing. 

More informative links that indicate what will happen, i.e. "Go to the My VA health site..."  reduces the need for opening external sites in new tabs, because it helps clarify/warn where you are going.  But, if an external link also contributes to data loss/user confusion, then a new tab it may be warranted. 

Oh boy, did that become a long winded response!  

My thoughts on this specific link are, if the user is going to lose their session data and be logged out of VA.gov (data loss and confusing), I would definitely open the Cerner site in a new tab.  If that is not the case, then a new tab isn't necessary, and the link label can carry the weight to let users know that will be taken to a different site.  

Very open to discussing this more!

**Comment by joshkimux** [comment]
@mnorthuis Thank you for the update, that definitely makes sense and is consistent with WCAG exceptions as well (provided "opens in new tab" is added to the link text 😄 )

**Comment by ajakabcin** [comment]
Hi @joshkimux - another 508 ticket for your review. Thanks in advance!

**Comment by joshkimux** [comment]
<img width="641" alt="Screen Shot 2021-10-04 at 5 04 08 PM" src="https://user-images.githubusercontent.com/14154792/135924605-18486c8a-ec28-4d0c-8e92-34add9a03421.png">

@erikphansen 👏 👏 👏
